[CHA-1585] add user deletion, cart clearing, email under traits (#4207)
segmentio/segment-docs: src/connections/destinations/catalog/sailthru-v2/index.md

Send [Identify](/docs/connections/spec/identify) calls to to create or update a Sailthru profile for any identified user on your site.

> Email address passed at top level:

Reviewer: ```suggestion
Pass an email address at the top level using the `identify` call:
```

});
```

> Email passed under context.traits:

Reviewer: ```suggestion
Pass an email under `context.traits` using the identify call:
```

```js
analytics.identify("assigned-userId", {
 "name": "Stephen Noel",
 "plan": "premium",
 "logins": 5
},
{
 "traits": {"email": "snoel@sailthru.com"}
}
);
```

> **Note:** Sailthru will also search for the email address in the identify call() under context.traits if not provided at the top-level.

Reviewer: ```suggestion
> note ""
> **NOTE:** Sailthru searches for the email address in the `identify` call under `context.traits` if it isn't provided at the top-level.
```

});
```

#### Delete a User

> For user deletion, the email needs to be included in the userId or context.traits track call, in a call like the following:

Reviewer: ```suggestion
To delete a user, the email needs to be included in the `userId` or `context.traits` track call, in a call like the following:
```

});
```

#### Delete a User

> To delete a user, the email needs to be included in the `userId` or `context.traits` track call, in a call like the following:

Reviewer: ```suggestion
To delete a user, you need to include the email in the `userId` or `context.traits` track call, in a call like the following:
```

#### Update Cart
Send `Product Added` and `Product Removed` Track events for Sailthru's abandoned cart messaging to enable the Cart Abandonment entry in Lifecycle Optimizer.

Sailthru abandoned cart messaging requires the `url` to function properly.

> The timestamp of the last payload received (purchase_incomplete.time) will be compared with the clearIncompleteCartAfterNHours value to determine if cart is expired. If the cart is determined to be expired, any pre-existing products in the cart will be removed.

Reviewer: ```suggestion
To determine if the cart is expired, the timestamp of the last payload received (`purchase_incomplete.time`) is compared with the `clearIncompleteCartAfterNHours` value. If the cart is expired, any pre-existing products in the cart will be removed.
```

  sku: 'G-32',
  category: 'Games',
  name: 'Monopoly: 3rd Edition',
  url: "https://www.example.com/product/monopoly",
  brand: 'Hasbro',
  variant: '200 pieces',
  price: 18.99,
  quantity: 1,
  coupon: 'MAYDEALS',
  position: 3,
  email: "xxxxxxx@example.com",
  clearIncompleteCartAfterNHours: 24
});
```

**Notes:**
* By default the value is null
* Requirement is for the value to be greater than 0

> * Only whole positive numbers are supported. If fractional numbers are included, the number will be truncated (e.g. 1.5 will be 1)

Reviewer: ```suggestion
* Only whole positive numbers are supported. If fractional numbers are included, the number will be truncated (for example, 1.5 will be 1)
```

> * If a number is passed in a string format such as "1", it will be parsed and converted to integer

Reviewer: ```suggestion
* If a number is passed in a string format such as `1`, it will be parsed and converted to an integer
```

Author: I changed it to `"1"`.  We'd like to keep the double quotes to make clear that this a is a string representation of a number, not a number literal.  I adopted your suggestion to use code formatting though.
